### DISCUSSION TITLE
WPS Feature

### DISCUSSION THREAD

**Discussion by costa-victor** [Moddable-OpenSource/moddable#883] [Ideas]
Hello there,

I've been studying and doing some network testing with a new router I bought, and I was thinking about bringing the WPS feature to Moddable, so I would like to know if there is any reason why this feature is not available? Or would it be a welcome feature?

I was thinking about bringing this feature to ESP32, but considering it can be used by other platforms. So I was looking at the existing approach to taking care of various platforms used by [Digital module](https://github.com/Moddable-OpenSource/moddable/tree/public/modules/io/digital).

Thanks,
Victor

**Comment by phoddie** [comment]
I don't use WPS and I'm unsure how widely used it is these days. Maybe someone else has a strong opinion on this. @wilberforce has done quite a bit of interesting work on Wi-Fi provisioning, for example.

Wi-Fi Protected Set-up (WPS) seems to have many options. Are you referring specifically to the [Push Button mode](https://en.wikipedia.org/wiki/Wi-Fi_Protected_Setup#Modes)?

**Reply by costa-victor** [reply]
I'm familiar with Push Button mode (PBC) and Personal Identification number (PIN), these two are available for [ESP-IDF](https://github.com/espressif/esp-idf/tree/master/examples/wifi/wps).

I have done some tests and projects using them, specifically the PBC seems to be an easy and cool option for provisioning.

Yes @wilberforce and @phoddie , i'm referring to the option the WPS button on the router.

Nice @wilberforce , I haven't know this tool.

**Comment by wilberforce** [comment]
I would have throught it might have been the push-button option on the router?

This is another area that might be worth investigation:
https://www.improv-wifi.com/

Proivisioning via serial for esp8266 and bluetooth for esp32
